airflow/providers/ssh/hooks/ssh.py

```diff
@@ -415,6 +415,9 @@ def _pkey_from_private_key(self, private_key: str, passphrase: Optional[str] = N
         :return: ``paramiko.PKey`` appropriate for given key
         :raises AirflowException: if key cannot be read
         """
+        if private_key.count("\n") < 2:
```

Comment:
```suggestion
        if private_key.split("\n", 2) < 2:
```

Using `count()` here would unnecessarily scan the entire key. It feels wrong to waste time for that since valid values are expected (which usually contains many lines).

Reply:
Good point, I've amended it accordingly.